docs/open-api/lineage.yaml

```diff
@@ -0,0 +1,281 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+---
+
+openapi: 3.0.2
+info:
+  title: OpenLineage
+  version: 2-0-2
+  description: OpenLineage is an open source **lineage and metadata collection API** for the data ecosystem.
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+paths:
+  /api/lineage:
```

Comment:
you should also add the path ref to openapi.yaml:
https://github.com/apache/gravitino/blob/c37e02c21e700b669a29a1d9bd6d13b59f0c67e9/docs/open-api/openapi.yaml#L50

Reply:
Another thought on this ...
The RESTful API design convention is to have plurals of resources in the path.
For the `POST` verb, most of the time it is about creating something.
So ....

1. I'm not sure if `lineage` is a suitable noun here for the resource collection, my guess is not. `lineage` can be an API prefix for differentiating this group of APIs from other APIs though.
2. By reading the JSON body payload, it smells to me that we are creating "events". However, the description of the method says that it is updating the run state of a job. This is confusing.
3. We have only one API for generating the "events", and we have no idea where those events are stored or processed. This API looks weird to me.

Reply:
@mchades adding the `lineage` path ref  in `openapi.yaml`.  @tengqm , this endpoint actually updates the lineage of the job by receiving run events which contains job lineage information.   This is the first step to support lineage in Gravitino, in this stage, we only provide an endpoint to receive OpenLineage run event and print the event in log file,   whether providing other endpoints like quering event depended on ther requirement of the community.

Reply:
The concern, from my side, is that for the sake of potential future extensions, should we use `/api/lineage/v1/events` as the endpoint?

Reply:
I prefer to keep the current endpoint. Because this is the final stage of the lineage support, we'd better not make big changes.

Reply:
The all too often cases are that once something is merged, it is hard to get it polished or removed. Time can be a concern, but I don't think we are in a hurry to get things in.

In fact, this won't be a disruptive change.

Reply:
@jerryshao do you think it is necessary to change the endpoint from `lineage` to `/lineage/v1/events`?